libs/utils/utils.js

```diff
@@ -686,6 +686,53 @@ export function convertStageLinks({ anchors, config, hostname, href }) {
   });
 }
 
+function addCircleLoader(elem) {
+  loadStyle('../../../../libs/styles/progress-circle.css');
+  const overlay = createTag('div', { class: 'overlay' });
+  const loader = createTag('div', { class: 'loader' });
+  loader.style.display = 'block';
+  overlay.append(loader);
+  elem.prepend(overlay);
+}
+
+function removeCircleLoader(elem) {
+  elem.querySelector('.overlay').style.display = 'none';
+  elem.querySelector('.loader').style.display = 'none';
+}
+
+function addBarLoader(elem) {
+  loadStyle('../../../../libs/styles/progress-bar.css');
+  const container = createTag('div', { class: 'progress-bar-container' });
+  const progressBar = createTag('div', { class: 'progress-bar' });
+
+  const label = createTag('div', { class: 'progress-label' });
+  label.textContent = 'Launching the app store...';
+  container.append(label);
+
+  const track = createTag('div', { class: 'progress-bar-value' });
+  track.style.display = 'block';
+  progressBar.append(track);
+  container.append(progressBar);
+  elem.replaceWith(container);
+  return container;
+}
+
+function removeBarLoader(elem, a) {
+  elem.replaceWith(a);
+}
+
+async function decorateQuickLink(a, hasConsent) {
+  if (!window.alloy) return;
+  const { getECID } = await import('../blocks/mobile-app-banner/mobile-app-banner.js');
```

Comment:
Importing the `mobile-app-banner` to get the ECID is an anti-pattern. You create a dependency on an encapsulated method, which also would lead to things such as the lana logs being wrong, which would hint the `mobile-app-banner` itself had an error, while it occurred in the utils.
```.catch((err) => window.lana.log(`Error fetching ECID: ${err}`, { tags: 'mobile-app-banner' }));```

Utils is consumed by everyone and thus changes should be done more carefully, blocks should just know about their own existence making refactoring isolated to testing the block alone and that's enough.

Reply:
Would it be a better solution to add the `getECID` function in utils and then import it in mobile app banner?

Reply:
Hmm 😬 That's a good question, we started guarding adding to utils quite a lot.

If it's used in 3-4+ blocks yes. If it's used twice so far - it might be banner to have the connection to branch for now. We should ensure to test performance though, so we don't add any render-blocking long import chains.